mapper/orm/src/main/java/org/hibernate/search/mapper/orm/batchindexing/impl/DefaultMassIndexerFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.batchindexing.impl;
+
+import java.util.Properties;
+
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.search.mapper.orm.MassIndexer;
+import org.hibernate.search.mapper.orm.batchindexing.spi.MassIndexerFactory;
+
+/**
+ * The {@link MassIndexer} implementation used when none is specified in the configuration.
+ *
+ * @author Davide D'Alto
+ */
+public class DefaultMassIndexerFactory implements MassIndexerFactory {
```

Comment:
Hmm, I didn't know about that.

Mass indexer factories don't make much sense until we provide a way for user to access internal metadata: what could they do with a session factory and a list of entity types?

I think we should remove the whole concept of "mass indexer factory" for now and just use `new MassIndexerImpl()` wherever this was used. We'll reintroduce factories later if necessary; I suspect it's only needed for OGM, so we'll have a look again when we start integrating with OGM.

Could do that in a separate commit, please?

Reply:
Sure